lockfile/src/parsers/yarn.rs

```diff
@@ -29,28 +35,80 @@ fn entry(input: &str) -> IResult<&str, Package> {
     Ok((i, my_entry))
 }
 
-fn parse_entry(input: &str) -> IResult<&str, Package> {
-    context("entry", tuple((entry_name, entry_version)))(input).map(|(next_input, res)| {
-        let (name, version) = res;
-        (next_input, Package {
-            name: name.to_string(),
-            version: PackageVersion::FirstParty(version.to_string()),
-            package_type: PackageType::Npm,
-        })
-    })
+fn parse_entry(input: &str) -> IResult<&str, Option<Package>> {
+    let (input, (name, version)) = context("entry", tuple((entry_name, entry_version)))(input)?;
+
+    let version = match version {
+        Some(version) => version,
+        None => return Ok((input, None)),
+    };
+
+    let package = Package { version, name: name.to_string(), package_type: PackageType::Npm };
+
+    Ok((input, Some(package)))
 }
 
 fn entry_name(input: &str) -> IResult<&str, &str> {
+    // Strip optional quotes.
     let (i, _) = opt(tag(r#"""#))(input)?;
```

Comment:
nit again (I know it wasn't changed in this PR), but any reason to use `i` and not shadow `input` like we do in most of the other parsing functions we have in this file?

Reply:
Happy to change this. I don't like this either.